libs/ui-toolkit/src/components/popover/popover.tsx

```diff
@@ -14,6 +15,12 @@ export const Popover = ({
   open,
   onChange,
 }: PopoverProps) => {
+  const popoverContent = useRef<HTMLDivElement>(null);
+  /**
+   * It's responsible for preventing auto-closing of the popover when
+   * `focusOutside` event is originated from within (e.g. opens dialog)
```

Comment:
question (non blocking): I'm surprised that this isn't the default behavior of the popover component. Are we perhaps doing something wrong somewhere?

Reply:
It should be like that but because there's not radix's trigger inside the popover it doesn't know it originated from within.

Reply:
But I agree with your comment that it adds complexity that we do not need at the moment. I'm happy to remove that part.